errors_test.go

```diff
@@ -118,3 +118,50 @@ func TestGetType(t *testing.T) {
 	}
 
 }
+
+func TestGetDetails(t *testing.T) {
+	foo := struct{ foo string }{"foo"}
+	bar := struct{ bar string }{"bar"}
+	baz := struct{ baz string }{"baz"}
+	tests := []struct {
+		err      error
+		expected []interface{}
+	}{
+		{nil, []interface{}{}},
+		{io.EOF, []interface{}{}},
+		{AddDetails(nil, foo), []interface{}{foo}},
```

Comment:
Please see my previous comment and add a test case `AddDetails(nil, nil)` here

Reply:
Done